docs/ldml/tr35-keyboards.md

```diff
@@ -308,6 +308,263 @@ For extensibility, the `<special>` element will be allowed at nearly every level
 
 See [Element special](tr35.md#special) in Part 1.
 
+## Normalization
+
+Normalization will not typically be the responsibility of the keyboard author, rather this will be managed by the keyboard implementation.
+The keyboard implementation will apply normalization as appropriate when matching transform rules and `<display>` value matching.
+Output from the keyboard, following application of all transform rules, will be normalized to implementation- or application-requested form.
+
+The attribute value `normalization="disabled"` can be used to indicate that no automatic normalization is to be applied in input, matching, or output.  Using this setting should be done with caution. See [`<settings>`](#element-settings).
+
+Keyboard source files may be in any normalization format, because they are processed in NFD for purposes of matching. However, authors should be aware of areas where normalization affects keyboard operation.
```

Comment:
These two paragraphs do read a little like they contradict each other.

```suggestion
The attribute value `normalization="disabled"` can be used to indicate that no automatic normalization is to be applied in input, matching, or output. Using this setting should be done with caution:

* When this attribute value is used, all matching and output uses only the exact codepoints provided by the keyboard author.
* Input context from the application may not be normalized, which means that the keyboard author should consider all possible combinations, including NFC, NFD, and mixed normalization in `<transform from=` attributes.
* See [`<settings>`](#element-settings) for further details.

The remainder of this section only applies when `normalization="disabled"` is not used.

Keyboard source files may be in any normalization format, because they are processed in NFD for purposes of matching. However, authors should be aware of areas where normalization affects keyboard operation.
```